src/parser/shared/modules/DebuffUptime.tsx

```diff
@@ -0,0 +1,43 @@
+import Enemies from 'parser/shared/modules/Enemies';
+import Analyzer from 'parser/core/Analyzer';
+import { ThresholdStyle } from 'parser/core/ParseResults';
+import { QualitativePerformance } from 'parser/ui/QualitativePerformance';
+
+export default class DebuffUptime extends Analyzer {
+  static dependencies = {
+    enemies: Enemies,
+  };
+  protected enemies!: Enemies;
+
+  debuffId!: number;
```

Comment:
the subclasses of this appear to be using `debuffSpell` instead of `debuffId`. I don't see them setting `debuffId`. I *think* this can be changed to:

```suggestion
  abstract readonly debuffSpell: Spell;
```

and then you can either add a getter for `debuffId` or use `this.debuffSpell.id`

Reply:
(alternately, this can be a constructor parameter. that is how i usually do it)

Reply:
updated